src/html5/parser.rs

```diff
@@ -3905,4 +3894,56 @@ mod test {
         binding.set_node_named_id(NodeId(4), "otherid");
         assert!(binding.get_node_by_named_id("myid").is_none());
     }
+
+    #[test]
+    fn tests2_dat_720() {
+        let document = parse("<!DOCTYPE html>X</html>X");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "XX"
+            "#,
+        );
+    }
+
+    #[test]
+    fn tests2_dat_731() {
+        let document = parse("<!DOCTYPE html>X</html> ");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "X "
+            "#,
+        );
+    }
+
+    #[test]
+    fn tests2_dat_754() {
+        let document = parse("<!DOCTYPE html>X<p/x/y/z>");
+
+        assert_same(
+            &document,
+            r#"
+            | <!DOCTYPE html>
+            | <html>
+            |   <head>
+            |   <body>
+            |     "X"
+            |     <p>
+            |       x=""
+            |       y=""
+            |       z=""            
+            "#,
```

Comment:
We need a stable key ordering in order to pass this and similar tests.

Reply:
i think we can order attributes alphabetically.. we might need to have a separate Vec<usize> for ordering besides the attribute hashmap (or have some kind of "ordered" hashmap that does this for us)

Reply:
I like not preprocessing the test input.  Perhaps we can parameterize this type for use in tests.  Ordering alphabetically incurs `O(n log(n))` overhead (for small n).